internal/provider/provider.go

```diff
@@ -57,31 +58,25 @@ func (p *VellumProvider) Configure(ctx context.Context, req provider.ConfigureRe
 		return
 	}
 
-	// Configuration values are now available.
-	// if data.Endpoint.IsNull() { /* ... */ }
-
-	// Example client configuration for data sources and resources
 	client := http.DefaultClient
 	resp.DataSourceData = client
 	resp.ResourceData = client
 }
 
 func (p *VellumProvider) Resources(ctx context.Context) []func() resource.Resource {
 	return []func() resource.Resource{
-		NewExampleResource,
+		document_index.NewDocumentIndexResource,
 	}
 }
 
 func (p *VellumProvider) DataSources(ctx context.Context) []func() datasource.DataSource {
 	return []func() datasource.DataSource{
-		NewExampleDataSource,
+		document_index.NewDocumentIndexDataSource,
 	}
 }
 
 func (p *VellumProvider) Functions(ctx context.Context) []func() function.Function {
-	return []func() function.Function{
-		NewExampleFunction,
-	}
+	return []func() function.Function{}
```

Comment:
@dsinghvi will we have a way to define terraform functions? I don't have any in mind right now, but could be nice to have in our back pocket